Star Seeker/Views/Compositions/MovementController/JoystickMovementController.swift

```diff
@@ -105,22 +130,123 @@ class JoystickMovementController : MovementController {
         )
     }
     
-    static func defaultDirectionIndicator ( to target: SKSpriteNode ) -> SKSpriteNode {
-        let arrowTexture                = SKTexture( image: UIImage(systemName: "arrowshape.up.fill")! )
-        let directionIndicator          = SKSpriteNode( texture: arrowTexture, size: arrowTexture.size() )
-            directionIndicator.isHidden = true
-            directionIndicator.size     = CGSize( width: 50, height: 50 )
-            directionIndicator.position = CGPoint( x: target.position.x, y: target.position.y + target.size.height )
-        
-        return directionIndicator
+    static func scaleBottomController (target: SKSpriteNode, to: Double) {
+        let scaleAction = SKAction.scale(to: to, duration: 0.2)
+        target.run(scaleAction, withKey: "scaleBottomController")
     }
     
-    /** Specifies how large the joystick knob will be */
-    let bSize   = UIConfig.SquareSizes.mini + 10
-    /** Convenience class-exclusive variable, which gets its value by deriving horizontal impulse value from GameConfig */
-    let hImpls   : CGFloat
-    /** Convenience class-exclusive variable, which gets its value by deriving vertical impulse value from GameConfig */
-    let vImpls   : CGFloat
+    static func fadeInOutNode (target: SKSpriteNode, isFadeIn: Bool) {
+        var action: SKAction
+        var key: String
+        if (isFadeIn) {
+            action = SKAction.fadeIn(withDuration: 0.2)
+            key = "fadeIn"
+        } else {
+            action = SKAction.fadeOut(withDuration: 0.2)
+            key = "fadeOut"
+        }
+        target.run(action, withKey: "\(key)\(String(describing: target.name))")
+    }
+    
+    static func defaultDirectionIndicator (impulse: CGVector, startingPosition: CGPoint) -> SKShapeNode {
+        let points = calculateTrajectoryPoints(impulse: impulse, startingPosition: startingPosition)
+        let dashedPath = createDashedPath(points: points, lineWidth: 2.0, dashLength: 10.0, gapLength: 5.0)
+        
+        let path = CGMutablePath()
+        path.addLines(between: points)
+        
+        let shapeNode = SKShapeNode(path: dashedPath)
+        shapeNode.strokeColor = .white
+        shapeNode.lineWidth = 3.0
+        shapeNode.zPosition = 20
+        
+        return shapeNode
+        
+    }
+    
+    static func createDashedPath(points: [CGPoint], lineWidth: CGFloat, dashLength: CGFloat, gapLength: CGFloat) -> CGPath {
```

Comment:
takes the given points, and draws a dashed line, according to to a certain dashLength and lineWidth

Reply:
correct